std/string.d

```diff
@@ -2656,26 +2656,107 @@ Range stripLeft(Range)(Range str)
 /++
     Strips trailing whitespace (as defined by $(XREF uni, isWhite)).
 
-    Returns: $(D str) stripped of trailing whitespace.
+    Params:
+        str = string or random access range of characters
 
-    Postconditions: $(D str) and the returned value
-    will share the same head (see $(XREF array, sameHead)).
+    Returns:
+        slice of $(D str) stripped of trailing whitespace.
   +/
-C[] stripRight(C)(C[] str) @safe pure @nogc
-    if (isSomeChar!C)
+auto stripRight(Range)(Range str)
+    if (isSomeString!Range ||
+        isRandomAccessRange!Range && hasLength!Range && hasSlicing!Range &&
+        isSomeChar!(ElementEncodingType!Range))
 {
-    import std.utf : codeLength;
-    foreach_reverse (i, dchar c; str)
+    alias C = Unqual!(ElementEncodingType!Range);
+    static if (isSomeString!Range)
     {
-        if (!std.uni.isWhite(c))
-            return str[0 .. i + codeLength!C(c)];
+        import std.utf : codeLength;
+        foreach_reverse (i, dchar c; str)
+        {
+            if (!std.uni.isWhite(c))
+                return str[0 .. i + codeLength!C(c)];
+        }
+
+        return str[0 .. 0];
     }
+    else
+    {
+        size_t i = str.length;
+        while (i--)
```

Comment:
why is there a need for a random access range? I see here only back() and popBack should be necessary